### PR TITLE
chore: Bump OpenFeature to 1.4.0 + react to deprecations

### PR DESCRIPTION
**Pull request by austindrenski**
See: open-feature/dotnet-sdk#154, open-feature/dotnet-sdk#170, open-feature/dotnet-sdk#171

### CODE REVIEW

File: build/Common.props [hunk 1]
```diff
@@ -25,7 +25,7 @@
     -->
     <MicrosoftSourceLinkGitHubPkgVer>[1.0.0,2.0)</MicrosoftSourceLinkGitHubPkgVer>
     <!-- 0.5+ -->
-    <OpenFeatureVer>[1.2,)</OpenFeatureVer>
+    <OpenFeatureVer>1.4.0</OpenFeatureVer>
```
Comment (askpt) [inline build/Common.props:28]:
```suggestion
    <OpenFeatureVer>[1.4,)</OpenFeatureVer>
```

Should we keep the 1.4 version as minor and let it bump naturally? Also, we might need to bump again when 2.0 comes with the breaking changes.

**Review by askpt — COMMENTED** [review]
Small question. LGTM otherwise.

File: build/Common.props (hunk 1 above)
Comment (toddbaert) [reply build/Common.props:28]: This makes sense to me

**Review by toddbaert — APPROVED** [review]

File: build/Common.props (hunk 1 above)
Comment (austindrenski) [reply build/Common.props:28]: IMO floating package versions haunt my nightmares. I've been burned more than once by a floating package version causing a lower version to get resolved from a CI build server cache.

That doesn't really apply in the era of cloud-hosted CI servers though, so will revert...

**Review by askpt — APPROVED** [review]
🚀

**Pull request merged**